ports/stm32/boards/make-pins.py

```diff
@@ -436,6 +436,13 @@ def print_adc(self, adc_num):
                     obj = "NULL"
                 print("  [{:d}] = {},".format(channel, obj))
             print("};")
+        else:
+            self.adc_table_size[adc_num] = 0
+            # If all of ADCx pins are hidden, print an empty table to avoid linker errors.
+            print("")
+            print("const pin_obj_t * const pin_adc{:d}[{:d}] = {{".format(adc_num, 0))
+            print("    /* No pins available for this table */")
+            print("}; ")
```

Comment:
To eliminate code duplication, might make sense to do something like:
```python
if adc_pins:
    table_size = max(adc_pins) + 1
else:
    table_size = 0
<existing code from the if-adc_pins block>
```

Reply:
Fixed and tested with all pins hidden, this is generated:
```
const pin_obj_t * const pin_adc1[0] = {
};

const pin_obj_t * const pin_adc2[0] = {
};

const pin_obj_t * const pin_adc3[0] = {
};
```